frontend/src/components/visual-editor/form/BlockFormProvider.tsx

```
@@ -23,7 +23,7 @@ function BlockFormProvider({
   methods,
   block,
 }: {
-  methods: UseFormReturn<IBlockAttributes, any, undefined>;
+  methods: UseFormReturn<IBlockAttributes>;
```

I've noticed that the useBlock hook doesn't handle the case where it's used outside of its provider. Since useContext(BlockContext) can return undefined in that scenario, I think the hook should explicitly check for this and throw an exception to avoid unexpected behavior. What do you think?

```
// current implementation 
export const useBlock = () => useContext(BlockContext);
```
```
// improved implementation 
export const useBlock = () => {
 const context = useContext(BlockContext);
 if(!context){
  throw new Error('useBlock must be used inside BlockContext.Provider');
 }
 return context;
}
```

---

Hi @abdou6666, a dedicated [issue](https://github.com/Hexastack/Hexabot/issues/911) is created for the mentioned update